### PR TITLE
MF-333:Carbonising OpenMRS Login Page

### PR DESCRIPTION
**Pull request by tendomart**
https://issues.openmrs.org/browse/MF-333

@brandones @FlorianRappl  @donaldkibet  Can you take sometime and look at this ?

### CODE REVIEW

File: src/root.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+@import "~carbon-components/scss/globals/scss/vendor/@carbon/elements/scss/type/styles";
```
Comment (FlorianRappl) [inline src/root.scss:1]: Wow these are quite some styles...

Do we need all of them? Usually they come already with the app shell... If not (and they are required for proper carbon usage) we should add them in the app shell...

File: package.json [hunk 2]
```diff
@@ -40,6 +40,11 @@
     "url": "https://github.com/openmrs/openmrs-esm-login/issues"
   },
   "dependencies": {
+    "@carbon/colors": "^10.19.0",
```
Comment (FlorianRappl) [inline package.json:43]: I don't see this being used.

File: package.json [hunk 3]
```diff
@@ -40,6 +40,11 @@
...
+    "@ibm/plex": "^5.1.3",
```
Comment (FlorianRappl) [inline package.json:44]: We only need the stylesheet of this, right? Any reason for using this stylesheet?

If you ask me we should rely on "pure" carbon and either duplicate (copy) or reimplement derived styles / components.

File: package.json [hunk 4]
```diff
@@ -40,6 +40,11 @@
...
+    "carbon-components": "^10.27.0",
```
Comment (FlorianRappl) [inline package.json:45]: This is one of the shared dependencies. No need to include it.

File: package.json [hunk 5]
```diff
@@ -40,6 +40,11 @@
...
+    "carbon-components-react": "^7.27.0",
+    "carbon-icons": "^7.0.7",
```
Comment (FlorianRappl) [inline package.json:47]: This is one of the shared dependencies. No need to include it.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (tendomart) [reply package.json:43]: yes cleaning this out , there was controversy  during selection.

File: package.json (hunk 5 above)
Comment (tendomart) [reply package.json:47]: @FlorianRappl  you mean  "carbon-icons": "^7.0.7" ?

Comment (FlorianRappl) [reply package.json:47]: Yes, always the line that I am commenting on.

File: package.json (hunk 3 above)
Comment (tendomart) [reply package.json:44]: I discovered Eric used ibm/ipex  at https://app.zeplin.io/project/5f7223cfda10f94d67cec6d0/screen/600805a8aca8ba86e8566ed3 for styling  but am removing it as well

File: package.json (hunk 2 above)
Comment (tendomart) [reply package.json:43]: Yes sure this is going to be remevod.

File: package.json (hunk 5 above)
Comment (tendomart) [reply package.json:47]: I was confused by the fact that it also included   `"carbon-components-react": "^7.27.0",`

File: src/root.scss (hunk 1 above)
Comment (tendomart) [reply src/root.scss:1]: @FlorianRappl  sorry for asking alot , which carbon dependencies are we supposed to use throughout esm modules ?
`carbon-components-react `or  `carbon-components` or **anything carbon** that get's the job done ?

**Comment by tendomart** [comment]
@FlorianRappl  done the necessary changes

File: src/root.scss (hunk 1 above)
Comment (FlorianRappl) [reply src/root.scss]: Here (https://github.com/openmrs/openmrs-esm-core/blob/master/packages/esm-app-shell/src/dependencies.ts) you see the shared dependencies. We share everything from carbon that is not React (or framework) specific. So `carbon-components-react` should be included, but `carbon-components` should appear only in `peerDependencies`.

File: src/login/login.component.tsx [hunk 6]
```diff
@@ -74,72 +75,71 @@ const Login: React.FC<LoginProps> = (props: LoginProps) => {
...
             />
-            <label htmlFor="username">
-              <Trans i18nKey="username">Username</Trans>
-            </label>
           </div>
           <div className="omrs-input-group">
-            <input
+            <TextInput.PasswordInput
               id="password"
+              invalidText="A valid value is required"
```
Comment (FlorianRappl) [inline src/login/login.component.tsx:97]: Translation?

File: src/login/login.component.tsx [hunk 7]
```diff
@@ -74,72 +75,71 @@ const Login: React.FC<LoginProps> = (props: LoginProps) => {
...
-            <input
+            <TextInput
               id="username"
               className={always("omrs-input-outlined").maybe(
                 "omrs-input-danger",
                 errorMessage
               )}
               type="text"
               name="username"
+              labelText="Username"
```
Comment (FlorianRappl) [inline src/login/login.component.tsx:86]: Translation?

File: package.json [hunk 8]
```diff
@@ -40,6 +40,7 @@
     "url": "https://github.com/openmrs/openmrs-esm-login/issues"
   },
   "dependencies": {
+    "carbon-components-react": "^7.27.0",
```
Comment (FlorianRappl) [inline package.json:43]: Great!

So now we only need to add `carbon-components` etc. to `peerDependencies` (shared dependencies go there, while bundled dependencies go into `dependencies`).

**Review by FlorianRappl — COMMENTED** [review]
Looks great - we are on the right track here. Thanks a lot! :beers:

**Comment by tendomart** [comment]
@FlorianRappl  is this still under review ?

**Comment by FlorianRappl** [comment]
No the review is done @tendomart 

See my comment on the package.json - shared packages should be added to the `peerDependencies` section.

**Comment by tendomart** [comment]
@FlorianRappl  yes i did that, i actually removed carbon-components , in the second commit.

**Comment by FlorianRappl** [comment]
Sorry @tendomart maybe I am blind, maybe the latest changes have not been pushed.

This is the *package.json* from your branch:

```json
{
  "name": "@openmrs/esm-login-app",
  "version": "3.0.2",
  "description": "The login microfrontend for the OpenMRS SPA",
  "main": "dist/openmrs-esm-login-app.js",
  "types": "src/index.ts",
  "license": "MPL-2.0",
  "homepage": "https://github.com/openmrs/openmrs-esm-login#readme",
  "scripts": {
    "start": "openmrs debug --run-project",
    "serve": "webpack-dev-server",
    "debug": "npm run serve",
    "build": "webpack --mode production",
    "lint": "eslint src --ext js,jsx,ts,tsx",
    "prettier": "prettier 'src/**' --write",
    "typescript": "tsc",
    "prepublishOnly": "npm run build",
    "test": "jest --config jest.config.json",
    "coverage": "npm test -- --coverage"
  },
  "browserslist": [
    "extends browserslist-config-openmrs"
  ],
  "keywords": [
    "openmrs"
  ],
  "publishConfig": {
    "access": "public"
  },
  "husky": {
    "hooks": {
      "pre-commit": "pretty-quick --staged && concurrently 'npm:lint' 'npm:test' 'npm:typescript'"
    }
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/openmrs/openmrs-esm-login.git"
  },
  "bugs": {
    "url": "https://github.com/openmrs/openmrs-esm-login/issues"
  },
  "dependencies": {
    "carbon-components-react": "^7.27.0",
    "kremling": "^2.0.3",
    "systemjs-webpack-interop": "^2.0.0"
  },
  "peerDependencies": {
    "@openmrs/esm-api": "3.x",
    "@openmrs/esm-config": "3.x",
    "@openmrs/esm-error-handling": "3.x",
    "@openmrs/esm-react-utils": "3.x",
    "i18next": "^19.6.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-i18next": "^11.7.0",
    "rxjs": "^6.5.4",
    "single-spa": "^4.4.1"
  },
  "devDependencies": {
    "@babel/core": "^7.9.0",
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/preset-env": "^7.9.0",
    "@babel/preset-react": "^7.9.4",
    "@babel/preset-typescript": "^7.9.0",
    "@openmrs/esm-api": "^3.1.6",
    "@openmrs/esm-config": "^3.1.6",
    "@openmrs/esm-error-handling": "^3.1.6",
    "@openmrs/esm-react-utils": "^3.1.6",
    "@testing-library/jest-dom": "^5.2.0",
    "@testing-library/react": "^9.5.0",
    "@types/jest": "^25.2.2",
    "@types/react": "^16.9.56",
    "@types/react-dom": "^16.9.5",
    "@types/react-router": "^5.1.7",
    "@types/react-router-dom": "^5.1.5",
    "@types/systemjs": "^6.1.0",
    "@types/webpack-env": "^1.15.1",
    "@typescript-eslint/parser": "^2.25.0",
    "babel-eslint": "^10.1.0",
    "babel-jest": "^26.0.1",
    "babel-loader": "^8.1.0",
    "babel-plugin-i18next-extract": "^0.6.1",
    "browserslist-config-openmrs": "^1.0.0",
    "clean-webpack-plugin": "^3.0.0",
    "concurrently": "^5.1.0",
    "css-loader": "^3.4.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.10.1",
    "eslint-config-ts-react-important-stuff": "^2.0.0",
    "eslint-plugin-prettier": "^3.1.2",
    "fork-ts-checker-webpack-plugin": "^4.1.1",
    "husky": "^4.2.3",
    "i18next": "^19.3.3",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^26.0.1",
    "jest-cli": "^26.0.1",
    "openmrs": "^3.1.6",
    "prettier": "^2.0.2",
    "pretty-quick": "^2.0.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-i18next": "^11.3.4",
    "react-router": "^5.1.2",
    "react-router-dom": "^5.1.2",
    "rxjs": "^6.5.4",
    "style-loader": "^1.1.3",
    "typescript": "^4.0.3",
    "webpack": "^4.43.0",
    "webpack-cli": "^3.3.11",
    "webpack-dev-server": "^3.11.0"
  }
}
```

I don't see `carbon-components` in the `peerDependencies`. I also don't see the `carbon-icons` in the `peerDependencies`. These are shared packages that are presumably used in `carbon-components-react`.

**Comment by tendomart** [comment]
@FlorianRappl  added as required.

**Review by FlorianRappl — APPROVED** [review]

**Pull request merged**